File: math/primenumbers/383-ransom-note/ransom-note.js
/**
 * @param {string} ransomNote
 * @param {string} magazine
 * @return {boolean}
 */
var canConstruct = function(ransomNote, magazine) {
    let arr=new Array(26).fill(0)

    for(let i=0;i<ransomNote.length;i++){
        arr[ransomNote.charCodeAt(i)-97]++
    }

  for(let i=0;i<magazine.length;i++){
       if(arr[magazine.charCodeAt(i)-97]!=0){
         arr[magazine.charCodeAt(i)-97]--
       }
    }

    for(let i=0;i<arr.length;i++){
        if(arr[i]!=0){
            return false
        }
    }

    return true
};

File: math/primenumbers/383-ransom-note/ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        
        int m=ransomNote.length();
        int n=magazine.length();

        int[] arr=new int[26];

        for(int i=0;i<n;i++){
            arr[magazine.charAt(i)-'a']++;
        }

        for(int i=0;i<m;i++){
            arr[ransomNote.charAt(i)-'a']--;
        }

        for(int i=0;i<arr.length;i++){
            if(arr[i]<0){
                return false;
            }
        }

        return true;

        
    }
}